alacritty/src/display/mod.rs

```diff
@@ -637,6 +634,14 @@ impl Display {
             config.window.dynamic_padding,
         );
 
+        // Queue renderer update if terminal dimensions/padding changed.
+        if new_size != self.size_info {
```

Comment:
You should compare after reserve lines I think, since it mutates terminfo, so it could be the same here, but reserve lines make it different.

See `self.size_info.reserve_lines(message_bar_lines + search_lines);`, it also mutates the `self.size_info`.